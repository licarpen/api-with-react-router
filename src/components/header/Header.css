.Header{
  font-family: Arial, Helvetica, sans-serif;
  margin: auto;
  text-align: center;
}