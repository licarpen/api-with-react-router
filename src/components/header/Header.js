import React from 'react';
import styles from './Header.css';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>
      Ryan and Marty Characters
    </h1>
    <nav>
      <ul>
        <li><Link to="/">Back to List</Link></li>
      </ul>
    </nav>
  </header>
);

export default Header;
