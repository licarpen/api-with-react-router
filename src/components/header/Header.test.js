import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import routeData from 'react-router';


describe('Header component', () => {
  const mockLocation = {
    pathname: '/character',
    hash: '',
    search: '',
    state: ''
  };
  
  beforeEach(() => {
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
  });

  it('matches a snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
