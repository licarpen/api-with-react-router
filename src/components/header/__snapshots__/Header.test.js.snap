// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches a snapshot 1`] = `
<header
  className="Header"
>
  <h1>
    Ryan and Marty Characters
  </h1>
  <nav>
    <Link
      to="/"
    >
      Back to List
    </Link>
  </nav>
</header>
`;
